### DISCUSSION TITLE
Is it possible to add and remove a module dynamically?

### DISCUSSION THREAD

**Discussion by pvonmoradi** [polybar/polybar#2688] [General]
I want to view network up/down transfer speeds. This is supported by network module. However, I don't want it to be always present in my bar as other modules have a 5 seconds refresh rate but the network module should have <1seconds refresh-rate for my case. This would incur many clock-cycles of processing effort on the machine. 

Is it possible to add and completely remove the module to a bar, programmatically? (hiding it isn't desired since it still burdens the CPU behind the scenes (I guess?))

**Comment by patrick96** [comment]
No that's not possible. Modules cannot be started/stopped dynamically. Hiding is the best you can do (which does continue to run and update the module in the background).

Alternatively you can use the [script module](https://github.com/polybar/polybar/wiki/Module:-script) and write a script that outputs the current network speed every < 1 seconds and implement some trigger to start/stop the automatic updates.
